pkg/build/gobuild.go

```diff
@@ -874,6 +880,14 @@ func (g *gobuild) buildOne(ctx context.Context, refStr string, base v1.Image, pl
 		if err != nil {
 			return nil, err
 		}
+
+		if g.sbomDir != "" {
+			fp := fmt.Sprintf("%s.%s", filepath.Join(g.sbomDir, appFilename(ref.Path())), g.sbomType)
+			log.Printf("Writing an SBOM to %s", fp)
+			if err := os.WriteFile(fp, sbom, 0644); err != nil {
+				return nil, err
+			}
+		}
```

Comment:
Can this be part of the sbomber impl's responsibility? It already knows the type of SBOM it's generating, and can also know its file name extension.

Reply:
yes, it can but wouldn't it cause code duplication? Because I need to add this logic into all of the somber impls. WDYT?

Reply:
I updated the code according to your feedback, can you please take a look? Thx